rpm/bundle-rust-npm.py

```diff
@@ -0,0 +1,251 @@
+    with open(spec_file, "r") as file:
+        contents = file.readlines()
+
+    log.info(f"Remove '{SPECFILE_COMMENT_LINE}' content from {spec_file}")
+    for line in contents:
+        if line == START_LINE:
+            remove_lines = True
+            log.debug(f"Remove '{START_LINE}' from {spec_file}")
+        elif line == END_LINE:
+            remove_lines = False
+            cleaned = True
+            log.debug(f"Remove '{END_LINE}' from {spec_file}")
+        elif not remove_lines:
+            result.append(line)
+        else:
+            log.debug(f"Remove '{line}' from {spec_file}")
+
+    with open(spec_file, "w") as file:
+        file.writelines(result)
+    return cleaned
+
+
+def write_provides_bundled(provides_lines: List[str], spec_file: str, cleaned: bool):
+    """Writes bundled package information to the spec file. 
+    Includes generated 'Provides' lines and marks the section for easy future modification.
+    """
+    # Find a line index where 'Provides' ends
+    with open(spec_file, "r") as file:
+        spec_file_lines = file.readlines()
+    last_provides = -1
+    for i in range(0, len(spec_file_lines)):
+        if spec_file_lines[i].startswith("%description"):
+            break
+        if spec_file_lines[i].startswith("Provides:"):
+            last_provides = i
+
+    # Insert the generated 'Provides' to the specfile
+    log.info(f"Add the fresh '{SPECFILE_COMMENT_LINE}' content to {spec_file}")
+    i = last_provides + 2
+    spec_file_lines.insert(i, START_LINE)
+    for line in sorted(provides_lines):
+        i = i + 1
+        log.debug(f"Adding '{line[:-1]}' as a line {i} to buffer")
+        spec_file_lines.insert(i, line)
+    i = i + 1
+    spec_file_lines.insert(i, END_LINE)
+
+    # Insert an empty line if we haven't cleaned the old content
+    # (as the old content already has an extra empty line that wasn't removed)
+    if not cleaned:
+        i = i + 1
+        spec_file_lines.insert(i, "\n")
+
+    log.debug(f"Commit the buffer to {spec_file}")
+    with open(spec_file, "w") as file:
+        file.writelines(spec_file_lines)
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+    log = setup_script_logger('bundle-rust-npm', args.verbose)
+
+    log.debug("389-ds-base Rust Crates and Node Modules to Bundled Downstream Specfile tool")
+    log.debug(f"Called with: {args}")
+
+    if args.backup_specfile:
+        backup_specfile(args.spec_file)
+
+    if not os.path.isdir(args.cargo_path):
+        log.error(f"Path {args.cargo_path} does not exist or is not a directory")
+        sys.exit(1)
+    if not os.path.isdir(args.npm_path):
+        log.error(f"Path {args.npm_path} does not exist or is not a directory")
+        sys.exit(1)
+
+    cleaned = clean_specfile(args.spec_file)
+
+    rust_output = run_cmd(["cargo", "license", "--json", "--current-dir", args.cargo_path])
+    npm_output = run_cmd(["license-checker", "--json", "--start", args.npm_path])
```

Comment:
See my comment above about installing packages globally. 
```suggestion
    npm_output = run_cmd(["npx", "license-checker", "--json", "--start", args.npm_path])
```